docs/admin/infrastructure/nublado/index.rst

```diff
@@ -0,0 +1,56 @@
+#######
+Nublado
+#######
+
+Although Nublado is itself a well-behaved Kubernetes application, in
+many environments it requires particular attention to the Kubernetes
+configuration settings in order to display the menu of available images
+correctly and to keep the prepuller from running continuously.
+
+Prepulled Images
+================
+
+The fundamental problem is that the Kubernetes setting
+``nodeStatusMaxImages`` is set to 50 by default.  The only way to
+retrieve a list of which images are present on a node is to query the
+node, and look through its ``status.images`` information.
+
+In general, our implementation relies on the supposition that prepulled
+images are the freshest, and that, eventually, people will stop using
+old images, and when the disk pressure garbage collection threshold is
+exceeded, the images that have not been used in the longest time will be
+purged.  When the ephemeral storage is sized such that there is not room
+for very many ``sciplat-lab`` images (but enough space to hold at least
+the full set that should be prepulled), and when ``sciplat-lab`` is the
+most common image found on nodes, this generally just works with no
+further attention: the menu stays populated with the current images, and
+since they are prepulled, they spawn quickly when selected.  Disk
+pressure cleans up outdated images, and everything works as it should.
+
+However, if the node has a large amount of ephemeral storage, and/or
+there is much non-Lab application use on the node, this can cause a
+problem for the prepuller: it is entirely possible for images that are
+indeed present on the node to not be in the first fifty images in the
+image list, and therefore not to be found when the prepuller determines
+which images need prepulling.
+
+This has two consequences: first, the prepuller will be constantly
+scheduling images as it prepulls the ones it wants, because even though
+the image is already resident on the node, the prepuller doesn't know
+that.  Second, these images, because the prepuller thinks they are not
+resident on all nodes, will not be visible in the JupyterHub spawner
+menu, although they will be available from the dropdown list.
+
+Fortunately there is a simple fix, provided one has access to kubelet
+configuration.  The configuration is described in
+https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/
+and includes the ``nodeStatusMaxImages`` setting.  The default value of
+50 should either be increased to something large enough that it's
+implausible that that many images would fit into ephemeral storage, or
+set to ``-1`` to remove the cap entirely.  While disabling the cap could
+in theory make node status extremely large, in practice, with the size
+of nodes we've been running RSP instances on, we have had at most
+hundreds, rather than thousands or millions, of container images on any
+given node.
```

Comment:
My advice here is to remember that documentation needs to be user-actionable. There's about 400 words and at the end of it there's nothing concretely actionable. The user still needs to do extra research both to figure out how to set the configuration, and how to decide what value to use, and how to know if the value is appropriate once it has been set.

I'd recommend breaking this section into two: "Setting nodeStatusMaxImages" and "Background".

That first section is where you tell the operator exactly what to do:

- Provide a command that can be run and link to documentation in Kubernetes if there's any set up required to run that command.
- Provide a specific value for nodeStatusMaxImages
- Indicate how the operator will know if the value is large enough; and what might happen if the value is too large.

In the section section, you can adapt the existing "Prepulled images" section.

Reply:
A couple Rubin style notes:

- Use sentence case for section titles
- Don't use "We" or "our;" they have no specific meaning in technical documentation
- Please use one-sentence-per-line wherever possible to make version control easier
- Put a title on URLs rather than pasting in a naked URL. See the earlier commend for how I added a re-usable link for `kubelet config`.